src/components/DatePicker/Common.tsx

```diff
@@ -0,0 +1,313 @@
+      <Icon name="calendar" />
+      <InputElement
+        data-testid="datepicker-input"
+        placeholder={placeholder}
+        readOnly
+        value={formattedSelectedDate}
+      />
+    </HighlightedInputWrapper>
+  );
+};
+
+// This is taken from InputElement from InputWrapper.tsx
+// We need to be able to change the color of some of the text of the 'input element'
+// but that isn't possible with an input element, so we have to fake it. Thus the name.
+const FakeInputElement = styled.div`
+  background: transparent;
+  border: none;
+  outline: none;
+  width: 100%;
+  color: inherit;
+  font: inherit;
+  ${({ theme }) => `
+    padding: ${theme.click.field.space.y} 0;
+    &::placeholder {
+      color: ${theme.click.field.color.placeholder.default};
+    }
+
+    &:disabled, &.disabled {
+      &::placeholder {
+      color: ${theme.click.field.color.placeholder.disabled};
+    }
+  `}
+`;
+
+const MutedColorSpan = styled.span`
+  color: ${({ theme }) => theme.global.color.text.muted};
+`;
+
+interface DateRangePickerInputProps {
+  isActive: boolean;
+  disabled: boolean;
+  id?: string;
+  placeholder?: string;
+  selectedEndDate?: Date;
+  selectedStartDate?: Date;
+}
+
+export const DateRangePickerInput = ({
+  isActive,
+  disabled,
+  id,
+  placeholder,
+  selectedEndDate,
+  selectedStartDate,
+}: DateRangePickerInputProps) => {
+  const defaultId = useId();
+
+  let formattedValue = <MutedColorSpan>{placeholder ?? ""}</MutedColorSpan>;
+  if (selectedStartDate) {
+    if (selectedEndDate) {
+      formattedValue = (
+        <span>
+          {selectedDateFormatter.format(selectedStartDate)} -{" "}
+          {selectedDateFormatter.format(selectedEndDate)}
+        </span>
+      );
+    } else {
+      formattedValue = (
+        <span>
+          {selectedDateFormatter.format(selectedStartDate)}{" "}
+          <MutedColorSpan>- end date</MutedColorSpan>
+        </span>
+      );
+    }
+  }
+
+  return (
+    <HighlightedInputWrapper
+      $isActive={isActive}
+      disabled={disabled}
+      id={id ?? defaultId}
+    >
+      <Icon name="calendar" />
+      <FakeInputElement data-testid="daterangepicker-input">
+        {formattedValue}
+      </FakeInputElement>
+    </HighlightedInputWrapper>
+  );
+};
+
+const weekdayFormatter = new Intl.DateTimeFormat(locale, { weekday: "short" });
+const headerDateFormatter = new Intl.DateTimeFormat(locale, {
+  month: "short",
+  year: "numeric",
+});
+
+const DatePickerContainer = styled(Container)`
+  background: ${({ theme }) =>
+    theme.click.datePicker.dateOption.color.background.default};
+`;
+
+const UnselectableTitle = styled.h2`
+  ${({ theme }) => `
+    color: ${theme.click.datePicker.color.title.default};
+    font: ${theme.click.datePicker.typography.title.default};
+  `}
+
+  user-select: none;
+`;
+
+const DateTable = styled.table`
+  border-collapse: separate;
+  border-spacing: 0;
+  font: ${({ theme }) => theme.typography.styles.product.text.normal.md}
```

Comment:
Missing semicolon? My IDE warns me, but CI seems to be ok with it 🤔

Reply:
I'll add it. I wonder if there's some kind of ASI happening in styled that makes this not break the CSS. Good find, regardless